### PR TITLE
[REEF-1743] Implement the Distributed Shell example in Java

### PR DESCRIPTION
**Pull request by motus**
JIRA: [REEF-1743](https://issues.apache.org/jira/browse/REEF-1743)

Closes #

### CODE REVIEW

File: lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/NumEvaluators.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * under the License.
+ */
+package org.apache.reef.examples.distributedshell;
+
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+
+/** Command line parameter: Number of evaluators to request. */
+@NamedParameter(doc = "Number of evaluators", short_name = "n", default_value = "1")
+public final class NumEvaluators implements Name<Integer> {
```
Comment (markusweimer) [inline lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/NumEvaluators.java]: Add a `private` constructor.

File: lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/RuntimeName.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * under the License.
+ */
+package org.apache.reef.examples.distributedshell;
+
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+
+/** Command line parameter: REEF runtime to use. Can be local, yarn, mesos, or hdi. */
+@NamedParameter(doc = "Runtime", short_name = "r", default_value = "local")
+public final class RuntimeName implements Name<String> {
```
Comment (markusweimer) [inline lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/RuntimeName.java]: Add a `private` constructor.

File: lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/ShellDriver.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import org.apache.reef.wake.EventHandler;
+import org.apache.reef.wake.time.event.StartTime;
+
+import javax.inject.Inject;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** The Driver code for REEF distributed shell application. */
+@Unit
+public final class ShellDriver {
```
Comment (markusweimer) [inline lang/java/reef-examples/src/main/java/org/apache/reef/examples/distributedshell/ShellDriver.java]: Does this class have to be `public`?

**Comment by motus** [comment]
@markusweimer We already have a HelloREEF unit test that runs locally and on YARN - do we need another test for pretty much identical functionality?

**Comment by markusweimer** [comment]
Fair enough. I was thinking of us breaking it. But we have more elaborate tests in the code base for that. Will merge.

**Pull request closed without merging**